keras/engine/training.py

```diff
@@ -888,10 +822,13 @@ def compile(self, optimizer, loss, metrics=None, loss_weights=None,
             else:
                 shape = self.internal_output_shapes[i]
                 name = self.output_names[i]
-                target = K.placeholder(ndim=len(shape),
-                                       name=name + '_target',
-                                       sparse=K.is_sparse(self.outputs[i]),
-                                       dtype=K.dtype(self.outputs[i]))
+                if K.is_placeholder(self.outputs[i]):
+                    target = K.placeholder(ndim=len(shape),
```

Comment:
This is the key location where type information is needed to address the `is_placeholder` problem @Dref360 brought up. If the model was initialized by a tf op which supplies data directly via `Input(tensor=tf_op)`, then there should not be a placeholder here. 

Perhaps proper flags can be determined based on how the layers were initialized and then set automatically in `model.compile()`? Please advise.